crypto/ctype.c

```diff
@@ -0,0 +1,556 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <e_os.h>
+#include "internal/ctype.h"
+
+#ifdef CHARSET_EBCDIC
+static const unsigned short ctype_char_map[256] = {
+   /*  0*/ CTYPE_MASK_ascii | CTYPE_MASK_cntrl,
+   /*  1*/ CTYPE_MASK_ascii | CTYPE_MASK_cntrl,
+   /*  2*/ CTYPE_MASK_ascii | CTYPE_MASK_cntrl,
+   /*  3*/ CTYPE_MASK_ascii | CTYPE_MASK_cntrl,
+   /*  4*/ 0,
```

Comment:
BTW, there are two EBCDIC tables in crypto/ebcdic. Well, there are three, one is fake to "emulate" EDCDIC encoding on conventional systems. But it might be sensible thing to do to make it work even with fake table...